src/pymor/discretizers/skfem/domaindiscretizer.py

```diff
@@ -0,0 +1,74 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright 2013-2021 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+from pymor.core.config import config
+
+if config.HAVE_SCIKIT_FEM:
+
+    import numpy as np
+    import skfem
+
+    from pymor.analyticalproblems.domaindescriptions import (RectDomain, CylindricalDomain, TorusDomain, LineDomain,
+                                                             CircleDomain, PolygonalDomain)
+
+    def discretize_domain(domain_description, diameter=1 / 100, mesh_type=None):
+
+        def discretize_RectDomain():
+            mt = mesh_type or skfem.MeshQuad
+            x0i = int(np.ceil(domain_description.width * np.sqrt(2) / diameter))
+            x1i = int(np.ceil(domain_description.height * np.sqrt(2) / diameter))
+            mesh = mt.init_tensor(
+                np.linspace(domain_description.domain[0, 0], domain_description.domain[1, 0], x0i + 1),
+                np.linspace(domain_description.domain[0, 1], domain_description.domain[1, 1], x1i + 1)
+            ).with_boundaries({
+                'left': lambda x: x[0] == domain_description.domain[0, 0],
+                'right': lambda x: x[0] == domain_description.domain[1, 0],
+                'top': lambda x: x[1] == domain_description.domain[1, 1],
+                'bottom': lambda x: x[1] == domain_description.domain[0, 1],
+            })
+
+            boundary_facets = {
+                bt: np.hstack([mesh.boundaries[edge]
+                              for edge in ['left', 'right', 'top', 'bottom']
+                              if getattr(domain_description, edge) == bt])
+                for bt in domain_description.boundary_types
+            }
+
+            return mesh, boundary_facets
+
+        def discretize_LineDomain():
+            mt = mesh_type or skfem.MeshLine
+            mesh = mt(
+                np.linspace(domain_description.domain[0], domain_description.domain[1],
+                            int(np.ceil((domain_description.domain[1] - domain_description.domain[0]) / diameter)))
+            ).with_boundaries({
+                'left': lambda x: x[0] == domain_description.domain[0],
+                'right': lambda x: x[0] == domain_description.domain[1]
+            })
+
+            boundary_facets = {
+                bt: np.hstack([mesh.boundaries[edge]
+                              for edge in ['left', 'right']
+                              if getattr(domain_description, edge) == bt])
+                for bt in domain_description.boundary_types
+            }
+
+            return mesh, boundary_facets
+
+        if isinstance(domain_description, RectDomain):
+            return discretize_RectDomain()
+        elif isinstance(domain_description, CylindricalDomain):
```

Comment:
Most of these are periodic domains if I understood correctly?

We have periodic meshes in `scikit-fem` (https://github.com/kinnala/scikit-fem/blob/master/docs/examples/ex42.py#L38) but those are still somewhat experimental feature and I have not heard of them being used outside of our docs.

Right now creating a doubly periodic square mesh is somewhat tedious because you must manually indentify the corner nodes to be eliminated. I made some snippet which solves eigenproblem of Laplacian on a doubly periodic square:
```python
In [1]: from skfem import *

In [2]: m = MeshQuad().refined(3)

In [14]: ix11 = m.nodes_satisfying(lambda x: (x[1] == 0) * (x[0] < 1) * (x[0] >
    ...: 0))

In [15]: ix22 = m.nodes_satisfying(lambda x: (x[1] == 1) * (x[0] < 1) * (x[0] >
    ...: 0))

In [42]: ix1 = m.nodes_satisfying(lambda x: (x[0] == 0) * (x[1] < 1) * (x[1] > 0
    ...: ))

In [43]: ix2 = m.nodes_satisfying(lambda x: (x[0] == 1) * (x[1] < 1) * (x[1] > 0
    ...: ))

In [44]: m.nodes_satisfying(lambda x: (x[0] == 0) * (x[1] == 0))
Out[44]: array([0])

In [45]: m.nodes_satisfying(lambda x: (x[0] == 1) * (x[1] == 0))
Out[45]: array([1])

In [46]: m.nodes_satisfying(lambda x: (x[0] == 1) * (x[1] == 1))
Out[46]: array([2])

In [47]: m.nodes_satisfying(lambda x: (x[0] == 0) * (x[1] == 1))
Out[47]: array([3])

In [48]: ix111 = np.array([1,2,3])

In [49]: ix222 = np.array([0,0,0])

In [50]: M = MeshQuad1DG.periodic(m, np.concatenate((ix1, ix11, ix111)), np.conc
    ...: atenate((ix2, ix22, ix222)))

In [51]: basis = Basis(M, ElementQuad1())

In [52]: A = laplace.assemble(basis)

In [53]: M = mass.assemble(basis)

In [54]: x = solve(A, M)

```
I guess defining these could use some improvements on our side. Let me know if you need any help or new features regarding this.